api/task_check_run.go

```diff
@@ -195,21 +195,37 @@ func ConvertToAdvisorDBType(dbType db.Type) (advisor.DBType, error) {
 }
 
 // IsSyntaxCheckSupported checks the engine type if syntax check supports it.
-func IsSyntaxCheckSupported(dbType db.Type) bool {
-	advisorDB, err := ConvertToAdvisorDBType(dbType)
-	if err != nil {
-		return false
+func IsSyntaxCheckSupported(dbType db.Type, mode common.ReleaseMode) bool {
+	if mode == common.ReleaseModeDev {
```

Comment:
I feel this should be mode == common.ReleaseModeDev || dbType == db.MySQL || dbType == db.TiDB

Reply:
updated